lib/puppet/provider/nssdb/certutil.rb

```diff
@@ -0,0 +1,72 @@
+require 'fileutils'
+
+Puppet::Type.type(:nssdb).provide(:certutil) do
+  commands :certutil => 'certutil'
+  commands :openssl => 'openssl'
+  commands :mkdir => 'mkdir'
+
+  def create
+    create_nssdb_dir
+    generate_password_file
+    generate_nssdb
+  end
+
+  def destroy
+    destroy_password_file
+    destroy_nssdb
+  end
+
+  def exists?
+    nssdb_exists?
+  end
+
+  private
+
+  def create_nssdb_dir
+    mkdir('-p', resource[:nssdb_dir])
+  rescue Puppet::ExecutionFailure => e
+    Puppet.err("Failed to create NSS database directory at #{resource[:nssdb_dir]}: #{e}")
+    return nil
+  end
+
+  def generate_password_file
+    openssl(
+      'rand',
+      '-base64', '24',
+      '-out', resource[:password_file]
+    )
+  rescue Puppet::ExecutionFailure => e
+    Puppet.err("Failed to generate new NSS database password file at #{resource[:password_file]}: #{e}")
+    return nil
+  end
+
+  def generate_nssdb
+    certutil(
+      '-N',
+      '-d', resource[:nssdb_dir],
+      '-f', resource[:password_file]
+    )
+  rescue Puppet::ExecutionFailure => e
+    Puppet.err("Failed to generate new NSS database at #{resource[:nssdb_dir]} with password file #{resource[:password_file]}: #{e}")
+    return nil
+  end
+
+  def destroy_password_file
+    FileUtils.rm_f resource[:password_file]
```

Comment:
```suggestion
    FileUtils.rm_f(resource[:password_file])
```

While what you have is correct, I think we should keep to a single style of calling functions